phpBB/assets/javascript/installer.js

```diff
@@ -0,0 +1,270 @@
+/**
+ * Installer's AJAX frontend handler
+ */
+
+(function($) { // Avoid conflicts with other libraries
+    // Installer variables
+    var pollTimer = null;
+    var nextReadPosition = 0;
+    var progressBarTriggered = false;
+    var progressTimer = null;
+    var currentProgress = 0;
+
+    // Template related variables
+    var $contentWrapper = $('.install-body').find('.main');
+
+    // Intercept form submits
+    intercept_form_submit($('#install_install'));
+
+    function poll_content(xhReq) {
+        var messages = xhReq.responseText;
+
+        do {
+            var unprocessed = messages.substring(nextReadPosition);
+            var messageEndIndex = unprocessed.indexOf('}\n\n');
```

Comment:
The variable is defined inside the do loop. However, you use it inside the while comparison. I'm pretty sure this might cause infinite loops since messageEndIndex is not defined outside the do {}.